test/Hintman/Suggest.hs

```diff
@@ -0,0 +1,57 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Hintman.Suggest (tests) where
+
+import Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import Relude
+import qualified Data.Text as T
+
+data LineState = Plain | Trailing
+
+space :: MonadGen m => m Char
+space = pure ' '
+
+genSpaces :: Int -> Int -> Gen Text
+genSpaces i j = Gen.text (Range.linear i j) space
+
+genPlain :: Gen Text
+genPlain =
+  T.intercalate " " <$>
+  Gen.list (Range.linear 0 20) (
+      Gen.text (Range.linear 5 100) Gen.alpha)
+
+genLines :: LineState -> Gen Text
+genLines Plain
```

Comment:
I feel like you should also have:

```haskell
genLineStates :: Gen [LineState]
```

So you then apply `genLine` generator for this list.